sections/Chapter 3 Sprint 1/postgreSQL.tex

```diff
@@ -2,33 +2,92 @@ \subsection{Relational Databases}
+In the coming sections, we will describe how one can model and describe a relational database design, both mathematically and with a more graphic design.
+After doing this, we will examine how operations on these models can be described using SQL (Section~\ref{sec:SQL}) and relational algebra (Section~\ref{sec:relationalAlgebra}).
 
-This abstraction is known as the logical layer and allows database administrators to manage the physical storage without directly manipulating the physical data representation.
-The logical layer describes what data are stored and the relationships between the data.
+\subsubsection*{Mathematical description}
+Relational database systems can be mathematically described using relations and sets, mapping a unique key to a tuple of information \cite[Chapter~2.3]{DBSBook}.
+The values of the tuples contained within the relation can be described by the attributes of the relation and their corresponding domains \cite{KatjaFirstPP}. 
+The relations are often described using a \textit{relational schema}, denoting the name and domains of the attributes.
 
-The highest of abstraction is the view layer which describes only part of the database. It exits to simplify the interaction with the system. Many views may exist for the same database.
+Equation \ref{eq:relational_schema} shows an example of a relation describing books as tuples containing three text fields and a positive integer.
+The relation also denotes a super key for the relation. A super key is one or more attributes that can uniquely identify a tuple in a given relation. Attributes describing the superkey are underlined.
 
-Entity-relation diagrams (ER diagrams) can be used to express the relationships in a relational database in a concise and graphical manner \cite{DBSBook} \cite{OracleRDBMS}.
\ No newline at end of file
+\begin{equation} \label{eq:relational_schema}
+    book(author\_name:text, title: text, number\_of\_pages:\mathbb{Z}^+, \underline{ISBN: text})
+\end{equation}
+Superkeys can be defined as $t_1 \in r,\neq t_2 \in r \implies t_1.K \neq t_2.K$. 
+
+That is, no two tuples $t_1, t_2$ from relation $r$ have same values for all superkey attributes $K$. 
+If the superkey does not contain extraneous attributes it is said to be \textit{minimal}. \cite[Chapter 2.3]{DBSBook}
+We will use the term \textit{primary key} to denote a chosen superkey of a relation. 
+When describing a database, it is often necessary to describe how various data is connected. 
+To do this, one can use \textit{foreign keys} to denote that tuples of $r_1$ is related to a tuple of $r_2$.
+
+
+One could model the relationship between a book owner and a book using the relational schemas seen in Equation \ref{eq:bookOwnerExample} and \ref{eq:relational_schema}.
+There, primary keys from other relations are used to reference unique tuples. The $owns$ relation describes how $book$ and $book\_owner$ are connected. 
```

Comment:
```suggestion
There, primary keys from other relations are used to reference unique tuples. The $owns$ relation describes how relations $book$ and $book\_owner$ are connected. 
```

Reply:
I think the first sentence "There, primary keys from other relations are used to reference unique tuples." needs to be rephrased such that it's more specific. It's not evident what exactly you are referring to.